### PR TITLE
Use code grant

### PR DESCRIPTION
**Pull request by craigatproperly**
Removes aws cognito SDK since the new system encapsulates that all in the changes of: 
https://github.com/GoProperly/auth_handler/pull/1

System now effectively just forwards to a login page and expects cookies to be set by the time  the browser returns to this page.  (Exception is that it is aware of an API it can use to to get a new jwt if the cookies indicate it is logged in, but, no current jwt is present in the cookies). 

Other Concerns to be reviewed or changed in a subsequent PR: 

Loss of deep linking capability: 

The system no longer carries deep linking through the login process, i.e. if you start logging in you don't come back to the same page.  However, the login is expected to happen infrequently. 

Inefficienty / Risk: 

Immediately after first login the client should be calling a "checkLogin" operation that lets it become aware of the userId it just logged in as (and links accounts on first use between the "properly user id" and the "oauth identifier".   With the login occurring on the back end the client is no longer in a position to execute it at the necessary and sufficient time, instead it calls it more frequently (and later) than it should be being called. Only expected consequences could be on switching browsers and with new users and would be limited to their first session.   Resolution will be to have an explicit location in the client that is redirected to.

### CODE REVIEW

File: lib/sessionaccount.js [hunk 1]
```diff
@@ -4,96 +4,41 @@ import * as URL from 'url-parse';
...
-  defaultReturnLogin: 'https://staging.admin.goproperly.com/',
-  defaultReturnSignout: 'https://staging.admin.goproperly.com/',
-  clientId: '3iuhnprmod0josge24ogarecfp',
-  loginHost: 'staging-goproperly-com.auth.us-east-1.amazoncognito.com',
-  loginResultRedirect: 'https://staging.admin.goproperly.com/hackloginhandlerbounce.html',
-  signoutResultRedirect: 'https://staging.admin.goproperly.com/hacksignoutbounce.html',
-  pool: STAGING_USER_POOL_ID,
-  checkLoginTemplate: 'https://2tpc2k1y2l.execute-api.us-east-1.amazonaws.com/staging/account/checklogin/%1$s',
-};
+//Cookie.get("foo")
```
Comment (sktse) [inline lib/sessionaccount.js]: Remove the commented out test code

**Comment by sktse** [comment]
> Immediately after first login the client should be calling a "checkLogin" operation that lets it become aware of the userId it just logged in as (and links accounts on first use between the "properly user id" and the "oauth identifier".

Since we are proxying the OAUTH login, we can just combine the two actions and return both IDs in the actual login response.
It's sort of the bridge between the login IDs and the relevant IDs to actually use our system.

**Comment by craigatproperly** [comment]
> Since we are proxying the OAUTH login, we can just combine the two actions and return both IDs in the actual login response on the server side.

Yeah, in that pattern I think I am going to need to carry the "requested" user_id through the login request as well (we generate userids before login, and ideally, we avoid changing them on first login) .   I am thinking the best approach is to tunnel a deep link and the requested userid through the state parameter. 

I was also thinking of a short cut to this rework being to just call checklogin and record a value in a cookie (since we are already using them) the first time the jwt is requested, and then skip checklogin if the value is present.

File: lib/sessionaccount.js (hunk 1 above)
Comment (craigatproperly) [reply lib/sessionaccount.js]: fixed

**Comment by craigatproperly** [comment]
Update: I have moved the performCheckLogin into a new location that will run immediately after trying to use the jwt on first login.   Its is a bit complex, but, I propose leaving it like this until we come back later and collapse the operation into one call as @sktse described.

**Pull request merged**